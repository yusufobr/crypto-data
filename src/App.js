import { useEffect } from 'react';
import './App.css';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { fetchCoins } from './redux/coin/coinsSlice';
import Coin from './components/Coin';
import Coins from './components/Coins';
import Navbar from './components/Navbar';

function App() {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchCoins());
  }, [dispatch]);

  return (
    <div>
      <BrowserRouter>
        <Navbar />
        <Routes>
          <Route index element={<Coins />} />
          <Route path="/coin/:coinId" element={<Coin />} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
