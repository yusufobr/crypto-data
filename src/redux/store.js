import { configureStore } from '@reduxjs/toolkit';
import coinsReducer from './coin/coinsSlice';

const store = configureStore({
  reducer: {
    coins: coinsReducer,
  },
});

export default store;
