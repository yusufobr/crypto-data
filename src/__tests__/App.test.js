import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import App from '../App';

describe('App.js', () => {
    it('should render correctly', () => {
        const tree = renderer.create(
            <Provider store={store}>
                <App />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})