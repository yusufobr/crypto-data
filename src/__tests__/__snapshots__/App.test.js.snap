// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js should render correctly 1`] = `
<div>
  <header
    className="container mx-auto px-2 py-6"
  >
    <nav
      className="text-3xl font-bold"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          src="logo.png"
          width={200}
        />
      </a>
    </nav>
  </header>
  <div
    className="container p-2 mx-auto mb-8 grid grid-cols-2 gap-4"
  >
    <div
      className="col-span-2 shadow-md mt-6"
    >
      <input
        className="w-full h-10 text-black p-2 rounded-lg"
        onChange={[Function]}
        placeholder="Search currency..."
        role="button"
        tabIndex={0}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
